upgrades/model.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package upgrades
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/version"
+)
+
+// minMajorUpgradeVersion defines the minimum version all models
+// must be running before a major version upgrade.
+var minMajorUpgradeVersion = map[int]version.Number{
+	3: version.MustParse("2.9.0"),
+}
+
+// UpgradeAllowed returns true if a major version upgrade is allowed
+// for the specified from and to versions.
+func UpgradeAllowed(from, to version.Number) (bool, version.Number, error) {
+	if from.Major >= to.Major {
```

Comment:
this should just be if they are equal

Reply:
unless we support downgrades?

Reply:
We do now support downgrade; it's been for minor versions eg 2.9 -> 2.8
But thinking about it, allowing a controller to be downgraded would be bad, so I'll fix, thanks.